### PR TITLE
Define available beatmap attributes

### PR DESCRIPTION
**Pull request by nanaya**
Verbose (duplicated attribute listing, cast definitions, and relation listing). There's benefit of validating attribute name though. The duplicated attributes can be removed but then it'll be back to returning null on invalid attribute access.

This also means faster relation getter 🤷‍♀️

Before:
```
Warming up... done
Run 1:   570 requests in 10.10s, 168.66MB read
Run 2:   605 requests in 10.21s, 179.01MB read
Run 3:   598 requests in 10.08s, 176.94MB read
Run 4:   596 requests in 10.10s, 176.35MB read
Run 5:   602 requests in 10.14s, 178.13MB read
Run 6:   593 requests in 10.09s, 175.46MB read
Run 7:   581 requests in 10.07s, 171.91MB read
Run 8:   582 requests in 10.05s, 172.21MB read
Run 9:   587 requests in 10.09s, 173.69MB read
Run 10:   587 requests in 10.07s, 173.69MB read
Final run:
Running 10s test @ https://on.nanaya.pro/beatmapsets/
  4 threads and 12 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   200.95ms   17.79ms 380.44ms   94.44%
    Req/Sec    14.62      5.40    30.00     64.66%
  593 requests in 10.10s, 175.46MB read
Requests/sec:     58.71
Transfer/sec:     17.37MB
```

After:
```
Warming up... done
Run 1:   630 requests in 10.10s, 186.41MB read
Run 2:   656 requests in 10.10s, 194.10MB read
Run 3:   650 requests in 10.09s, 192.33MB read
Run 4:   658 requests in 10.10s, 194.70MB read
Run 5:   658 requests in 10.09s, 194.70MB read
Run 6:   646 requests in 10.07s, 191.15MB read
Run 7:   630 requests in 10.11s, 186.41MB read
Run 8:   648 requests in 10.09s, 191.74MB read
Run 9:   649 requests in 10.08s, 192.03MB read
Run 10:   656 requests in 10.12s, 194.10MB read
Final run:
Running 10s test @ https://on.nanaya.pro/beatmapsets/
  4 threads and 12 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   185.32ms   25.43ms 520.20ms   96.45%
    Req/Sec    15.91      6.11    30.00     66.93%
  646 requests in 10.07s, 191.15MB read
Requests/sec:     64.15
Transfer/sec:     18.98MB
```

### CODE REVIEW

**Comment by peppy** [comment]
Would be helpful if you can include any profiling results in the PR description.

File: app/Models/Beatmap.php [hunk 1]
```diff
@@ -278,19 +203,71 @@ public function scoresBestMania()
...
+            'countTotal',
+            'diff_approach',
+            'diff_drain',
+            'diff_overall',
+            'filename',
+            'hit_length',
+            'passcount',
+            'playcount',
+            'playmode',
+            'score_version',
```
Comment (notbakaneko) [inline app/Models/Beatmap.php:234]: should add this to the list of `@property`

File: app/Models/Beatmapset.php [hunk 2]
```diff
@@ -926,6 +915,94 @@ public function language()
...
+            'approved_date_json',
+            'cover_updated_at_json',
+            'deleted_at_json',
+            'last_update_json',
+            'queued_at_json',
+            'submit_date_json',
+            'thread_icon_date_json' => $this->getJsonTimeFast($key),
+
+            'active',
+            'comment_locked',
```
Comment (notbakaneko) [inline app/Models/Beatmapset.php:973]: add class doc `@property`

File: app/Models/Beatmapset.php [hunk 3]
```diff
@@ -926,6 +915,94 @@ public function language()
...
+            'storyboard',
+            'video' => (bool) $this->getRawAttribute($key),
+
+            'artist_unicode' => $this->getArtistUnicode(),
+            'commentable_identifier' => $this->getCommentableIdentifierAttribute(),
+            'title_unicode' => $this->getTitleUnicode(),
+
+            'beatmapDiscussions',
+            'bssProcessQueues',
+            'allBeatmaps',
```
Comment (notbakaneko) [inline app/Models/Beatmapset.php:988]: b after a? 🤔

**Review by notbakaneko — CHANGES_REQUESTED** [review]
seems to be working, otherwise 👀 
(and about 5% faster for me despite it being a magic switch statement 👀 )

**Review by notbakaneko — APPROVED** [review]
:shipit:

**Pull request merged**